src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -461,12 +461,14 @@ const DataProcessingPage = ({ experimentId, experimentData }) => {
                         disabled={prefiltered}
                         data-testid='enableFilterButton'
                         onClick={async () => {
-                          dispatch(saveProcessingSettings(experimentId, currentStep.key));
-                          if (!processingConfig.meta.saveSettingsError) {
-                            dispatch(setQCStepEnabled(
-                              currentStep.key, !stepEnabled,
-                            ));
-                          }
+                          dispatch(saveProcessingSettings(experimentId, currentStep.key))
+                            .then(() => {
```

Comment:
Instead of this, can't you do `await dispatch(...)` to ensure that the next block of code is ran after the dispatch? This code block is inside an `async` function, so we should be able to `await`.